[package] qt/6.11: "Could not find executable lprodump"

### Description

When building Qt 6.11(.1), we get the following error:

```
ERROR: qt/6.11.1: Error in package() method, line 934
	assert False, f"Could not find executable {target}{extension} in {self.package_folder}"
	AssertionError: Could not find executable lprodump in /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p
```

This did not happen with Qt 6.10.

**EDIT:** Found the culprit in the [6.11 Qt Linguist release notes](https://doc.qt.io/qt-6/whatsnew611.html#qt-linguist): "Removed lprodump executable. Its functionality is now available via the `lupdate-pro` `-dump-json` option."

### Package and Environment Details

**macOS**

```
version: 2.28.1
conan_path: /opt/homebrew/bin/conan
python
  version: 3.14.5
  sys_version: 3.14.5 (main, May 10 2026, 10:21:34) [Clang 21.0.0 (clang-2100.0.123.102)]
  sys_executable: /opt/homebrew/Cellar/conan/2.28.1_2/libexec/bin/python
  is_frozen: False
  architecture: arm64
system
  version: Darwin Kernel Version 25.4.0: Thu Mar 19 19:33:09 PDT 2026; root:xnu-12377.101.15~1/RELEASE_ARM64_T8112
  platform: macOS-26.4-arm64-arm-64bit-Mach-O
  system: Darwin
  release: 25.4.0
  cpu: arm
```

**Windows**

```
version: 2.28.1
conan_path: c:\users\gitlabrunner\.local\bin\conan
python
  version: 3.14.3
  sys_version: 3.14.3 (tags/v3.14.3:323c59a, Feb  3 2026, 16:04:56) [MSC v.1944 64 bit (AMD64)]
  sys_executable: C:\Users\gitlabrunner\pipx\venvs\conan\Scripts\python.exe
  is_frozen: False
  architecture: AMD64
system
  version: 10.0.26200
  platform: Windows-11-10.0.26200-SP0
  system: Windows
  release: 11
  cpu: AMD64 Family 25 Model 17 Stepping 1, AuthenticAMD
```

### Conan profile

**macOS**

```
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=20
compiler.libcxx=libc++
compiler.version=21
os=Macos
os.version=13.3

[options]
openssl/*:openssldir=/etc/ssl/
qt/*:shared=True
qt/*:essential_modules=False
qt/*:qtsvg=True
qt/*:qttranslations=True
qt/*:qttools=True
qt/*:with_freetype=False
qt/*:with_harfbuzz=False
qt/*:with_sqlite3=False
qt/*:with_pq=False
qt/*:with_odbc=False
qt/*:with_brotli=False
qt/*:with_openal=False
qt/*:with_md4c=False

[tool_requires]
*: cmake/[>=3.28]

[conf]
tools.build:compiler_executables={'c': 'clang', 'cpp': 'clang++'}
tools.cmake.cmaketoolchain:generator=Ninja
qt/*:tools.cmake.cmaketoolchain:extra_variables={'QT_NO_XCODE_MIN_VERSION_CHECK': 'ON'}
```

**Windows**

```
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows

[options]
qt/*:shared=True
qt/*:essential_modules=False
qt/*:qtsvg=True
qt/*:qttranslations=True
qt/*:qttools=True
qt/*:with_freetype=False
qt/*:with_harfbuzz=False
qt/*:with_sqlite3=False
qt/*:with_pq=False
qt/*:with_odbc=False
qt/*:with_brotli=False
qt/*:with_openal=False
qt/*:with_md4c=False

[tool_requires]
*: cmake/[>=3.28]

[conf]
tools.build:defines=['_WIN32_WINNT=0x0A00']
```

### Steps to reproduce

`conan install --requires qt/6.11.1`

### Logs

<details><summary><b>macOS log</b></summary>

```
qt/6.11.1: Sources downloaded from 'conancenter'
qt/6.11.1: Calling source() in /Users/gitlab/.conan2/p/qt841db383ad6b9/s/src
qt/6.11.1: Downloading 1.02GB qt-everywhere-src-6.11.1.tar.xz
qt/6.11.1: Uncompressing qt-everywhere-src-6.11.1.tar.xz to /Users/gitlab/.conan2/p/qt841db383ad6b9/s/src
qt/6.11.1: apply_conandata_patches(): No patches defined in conandata
-------- Installing package qt/6.11.1 (21 of 21) --------
qt/6.11.1: Building from source
qt/6.11.1: Package qt/6.11.1:f67d44f8c414a530aaff06a9cd1c23f47c44d150
qt/6.11.1: settings: os=Macos os.version=13.3 arch=armv8 compiler=apple-clang compiler.cppstd=20 compiler.libcxx=libc++ compiler.version=21 build_type=Release
qt/6.11.1: options: device=None disabled_features= gui=True multiconfiguration=False opengl=desktop openssl=True qt3d=False qt5compat=False qtactiveqt=False qtcanvaspainter=False qtcharts=False qtcoap=False qtconnectivity=False qtdatavis3d=False qtdeclarative=False qtdoc=False qtgraphs=False qtgrpc=False qthttpserver=False qtimageformats=False qtlanguageserver=False qtlocation=False qtlottie=False qtmqtt=False qtmultimedia=False qtnetworkauth=False qtopcua=False qtopenapi=False qtpositioning=False qtquick3d=False qtquick3dphysics=False qtquickeffectmaker=False qtquicktimeline=False qtremoteobjects=False qtscxml=False qtsensors=False qtserialbus=False qtserialport=False qtshadertools=False qtspeech=False qtsvg=True qttasktree=False qttools=True qttranslations=True qtvirtualkeyboard=False qtwayland=False qtwebchannel=False qtwebengine=False qtwebsockets=False qtwebview=False shared=True widgets=True with_brotli=False with_dbus=False with_doubleconversion=True with_freetype=False with_glib=False with_gssapi=False with_harfbuzz=False with_libjpeg=False with_libpng=True with_md4c=False with_mysql=False with_odbc=False with_pcre2=True with_pq=False with_sqlite3=False with_vulkan=False with_zstd=False
qt/6.11.1: requires: openssl/3.6.2#36f92686a9285adfe2ccd1c1dbf9ea35:2ad44813339fc03d8c4803c09b127fbc5c17c073 pcre2/10.42#9a35f5089feb875ec61a38eca364ce77:bfd5464e8af37e04dbc1977d5f693dddc5e8f2e0 bzip2/1.0.8#c470882369c2d95c5c77e970c0c7e321:17790dca25f875411aab1d9ea2b73cf546ae423c double-conversion/3.3.0#6ad026cf92c316c476919f7f152072d7:d43d7dc8d99ecc55008328e7c24795d5ff3f5e96 libpng/1.6.58#19cb72905ae54f54948401f753faa2c1:83484d2c765a73130b337f625e9fca22fe6b1692 zlib/1.3.2#1cb806da49011867778ffb6ac7190fcb:cdf743a840ca00b695049cd02d5b5f324fb8151c opengl/system
qt/6.11.1: Building your package in /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b
qt/6.11.1: Calling generate()
qt/6.11.1: Generators folder: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release/generators
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'ZLIBConfig.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'module-ZLIB-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'double-conversion-config.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'double-conversion-config-version.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'OpenSSL-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'FindOpenSSL.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'module-PNG-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'BZip2ConfigVersion.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'module-BZip2-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'PNGConfigVersion.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'ZLIBConfigVersion.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'OpenSSLConfigVersion.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'BZip2-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'module-OpenSSL-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'FindBZip2.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'opengl_system-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'conandeps_legacy.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'PCRE2Config.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'double-conversion-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'OpenSSLConfig.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'PNG-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'PCRE2ConfigVersion.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'opengl_system-config.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'FindZLIB.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'opengl_system-config-version.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'BZip2Config.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'PCRE2-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'ZLIB-release-armv8-data.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'FindPNG.cmake' file.
qt/6.11.1: WARN: replace_in_file didn't find pattern ' IMPORTED)
' in 'PNGConfig.cmake' file.
qt/6.11.1: CMakeToolchain generated: conan_toolchain.cmake
qt/6.11.1: CMakeToolchain generated: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release/generators/CMakePresets.json
qt/6.11.1: CMakeToolchain generated: /Users/gitlab/.conan2/p/qt841db383ad6b9/s/src/CMakeUserPresets.json
qt/6.11.1: Generating aggregated env files
qt/6.11.1: Generated aggregated env files: ['conanbuild.sh']
qt/6.11.1: Calling build()
qt/6.11.1: Running CMake.configure()
qt/6.11.1: RUN: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="/Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/gitlab/.conan2/p/b/qt1e589b4e78693/p" -DCMAKE_TRY_COMPILE_CONFIGURATION="Release" -DQT_USE_VCPKG="OFF" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/gitlab/.conan2/p/qt841db383ad6b9/s/src"
-- Using Conan toolchain: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = ON
-- The CXX compiler identification is AppleClang 21.0.0.21000099
-- The C compiler identification is AppleClang 21.0.0.21000099
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The ASM compiler identification is AppleClang
-- Found assembler: /usr/bin/clang
-- Checking dependencies of submodule 'qtbase'
-- Checking dependencies of submodule 'qtsvg'
-- Checking dependencies of submodule 'qttools'
-- Skipping optional dependency 'qtactiveqt' of 'qttools', because building 'qtactiveqt' was explicitly disabled.
-- Skipping optional dependency 'qtdeclarative' of 'qttools', because building 'qtdeclarative' was explicitly disabled.
-- Checking dependencies of submodule 'qttranslations'
-- Configuring submodule 'qtbase'
-- [QtBase] CMAKE_BUILD_TYPE was already explicitly set to: 'Release'
-- Configuring submodule 'qtsvg'
-- Configuring submodule 'qttools'
CMake Warning (dev) at qtbase/cmake/QtPublicSbomHelpers.cmake:1963 (cmake_parse_arguments):
  The PACKAGE_VERSION keyword was followed by an empty string or no value at
  all.  Policy CMP0174 is not set, so cmake_parse_arguments() will unset the
  arg_PACKAGE_VERSION variable rather than setting it to an empty string.
Call Stack (most recent call first):
  qtbase/cmake/QtSbomHelpers.cmake:28 (_qt_find_package_extend_sbom)
  qttools/configure.cmake:10 (qt_find_package_extend_sbom)
  qtbase/cmake/QtModuleHelpers.cmake:751 (include)
  qttools/src/global/CMakeLists.txt:8 (qt_internal_add_module)
This warning is for project developers.  Use -Wno-dev to suppress it.
CMake Deprecation Warning at qttools/src/assistant/qlitehtml/src/3rdparty/litehtml/src/gumbo/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.
  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
-- Configuring submodule 'qttranslations'
-- Configuration summary has been written to /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release/config.summary
-- 
Qt is now configured for building. Just run 'cmake --build . --parallel'
Once everything is built, you must run 'cmake --install .'
Qt will be installed into '/Users/gitlab/.conan2/p/b/qt1e589b4e78693/p'
To configure and build other Qt modules, you can use the following convenience script:
        /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p/bin/qt-configure-module
If reconfiguration fails for some reason, try removing 'CMakeCache.txt' from the build directory
Alternatively, you can add the --fresh flag to your CMake flags.
-- Configuring done (36.0s)
-- Generating done (2.1s)
-- Build files have been written to: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release
qt/6.11.1: Running CMake.build()
qt/6.11.1: RUN: cmake --build "/Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release" -- -j8
[...]
[2385/2385] Generating ../../../qtbase/translations/qtwebsockets_zh_CN.qm
qt/6.11.1: Package 'f67d44f8c414a530aaff06a9cd1c23f47c44d150' built
qt/6.11.1: Build folder /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release
qt/6.11.1: Generating the package
qt/6.11.1: Packaging in folder /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p
qt/6.11.1: Calling package()
qt/6.11.1: Running CMake.install()
qt/6.11.1: RUN: cmake --install "/Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release" --prefix "/Users/gitlab/.conan2/p/b/qt1e589b4e78693/p"
-- Install configuration: "Release"
-- Installing: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p/lib/cmake/Qt6BuildInternals/Qt6BuildInternalsConfig.cmake
[...]
-- Installing: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p/translations/catalogs.json
-- Starting SPDX SBOM generation in build dir: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/b/build/Release/qttranslations/qt_sbom/staging-qttranslations.spdx.in
-- Finalizing SBOM generation in install dir: /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p/sbom/qttranslations-6.11.1.spdx
ERROR: qt/6.11.1: Error in package() method, line 934
	assert False, f"Could not find executable {target}{extension} in {self.package_folder}"
	AssertionError: Could not find executable lprodump in /Users/gitlab/.conan2/p/b/qt1e589b4e78693/p
```
</details>

<details><summary><b>Windows error</b></summary>

```
ERROR: qt/6.11.1: Error in package() method, line 934
	assert False, f"Could not find executable {target}{extension} in {self.package_folder}"
	AssertionError: Could not find executable lprodump.exe in C:\Users\gitlabrunner\.conan2\p\b\qt67b6e76cf15a4\p
```
</details>

--- a/recipes/qt/6.x.x/conanfile.py
+++ b/recipes/qt/6.x.x/conanfile.py
@@ -909,8 +909,10 @@ class QtConan(ConanFile):
                 targets.extend(["qhelpgenerator"])
             if "linguist" not in disabled_features:
                 targets.extend(["lconvert", "lrelease", "lrelease-pro", "lupdate", "lupdate-pro"])
-                if Version(self.version) < "6.11.0":
+                if ver < "6.11.0":
                     targets.extend(["lprodump"]) # Removed: https://doc.qt.io/qt-6/whatsnew611.html#qt-linguist
+                if ver >= "6.11.0":
+                    targets.extend(["lcheck", "ltext2id"]) # Added: https://doc.qt.io/qt-6/whatsnew611.html#qt-linguist
         if self.options.qtshadertools:
             targets.append("qsb")
         if self.options.qtdeclarative:

--- a/recipes/qt/6.x.x/conanfile.py
+++ b/recipes/qt/6.x.x/conanfile.py
@@ -908,7 +908,9 @@ class QtConan(ConanFile):
                 # and `qhelpgenerator` is a subdirectory of assistant in qttools
                 targets.extend(["qhelpgenerator"])
             if "linguist" not in disabled_features:
-                targets.extend(["lconvert", "lprodump", "lrelease", "lrelease-pro", "lupdate", "lupdate-pro"])
+                targets.extend(["lconvert", "lrelease", "lrelease-pro", "lupdate", "lupdate-pro"])
+                if Version(self.version) < "6.11.0":
+                    targets.extend(["lprodump"]) # Removed: https://doc.qt.io/qt-6/whatsnew611.html#qt-linguist
         if self.options.qtshadertools:
             targets.append("qsb")
         if self.options.qtdeclarative:

